File: controllers/Camera.js
class Camera {
  captureImage(callback) {
    console.log("in capture...");
    var PiCamera = require("pi-camera");

    var camera = new PiCamera({
      mode: "photo",
      output: `./Data/imgs/img.jpg`,
      width: 640,
      height: 480,
      nopreview: true
    });

    camera
      .snap()
      .then(result => {
        //Img captured
        console.log("...captured");
        console.log(result);
        callback(true);
      })
      .catch(error => {
        // Handle your error
        console.log("...error");
        console.log(error);
        callback(false);
      });
  }
}
module.exports = Camera;

File: src/controllers/camera.py
from picamera import PiCamera
import datetime


class camera_control(object):
    def __init__(self):
        self.camera = PiCamera()
        self.camera.resolution = (1024, 768)
        self.camera.start_preview()

    def snap(self, callback):
        self.camera.capture("img.jpg")
        print(datetime.datetime.now())
        callback("stuff")

